### PR TITLE
Fix & Improve Tag in-code documentation

### PR DESCRIPTION
**Pull request by Ktar5**
I fixed a misnamed method in the code samples and also cleaned them up, added some more comments, and tried to make it more understandable after I had some trouble learning them myself. Hopefully this will benefit future users coming to this page.

### CODE REVIEW

File: docs/utilities/tags.md [hunk 1]
```diff
@@ -21,17 +21,45 @@ Using Tags In Code
...
+// Tags work through Items, not ItemStacks so we need to get the Item from our ItemStack
 Item unknownItem = stack.getItem();
-boolean isInGroup = ItemTags.getCollection().getOrCreateTag(myTagId).contains(unknownItem);
-// alternatively, can use getTag and perform a null check
+// This line will either retrieve the Tag (referenced by our ResourceLocation) or create it if it is not
+// in the ItemTags collection of all the tags. Then it will check if our Item (again, not ItemStack)
+// exists in the set of Items that belong to this Tag
+boolean isInGroup = ItemTags.getCollection().getOrCreate(myTagId).contains(unknownItem);
+// Alternatively, you can use the getTag and perform a null check to see if the tag exists and do it that way
+Tag retrievedTag = ItemTags.getCollection.get(myTagId);
```
Comment (Alpvax) [inline docs/utilities/tags.md]: Typo: missing parentheses.
I believe that should be getCollection()

Comment (Ktar5) [reply docs/utilities/tags.md]: Indeed! Changing it right now.

**Comment by Ktar5** [comment]
Any update on this?

File: docs/utilities/tags.md [hunk 2]
```diff
@@ -21,17 +21,45 @@ Using Tags In Code
...
 Block, Item, and Fluid tags are automatically sent from the server to any remote clients on login and reload. Function tags are not synced.
 
+Tags can be accessed in code through the `Tags.Items` and `Tags.Blocks` classes for all the Vanilla Minecraft tags, and through `ResourceLocation`s for ForgeTags.
+See the examples below for how to use each.
+
 `BlockTags.getCollection()` and `ItemTags.getCollection()` will retrieve the current `TagCollection`, from which you can retrieve a `Tag` object by its ID.
 With a `Tag` object in hand, membership can be tested with `tag.contains(thing)`, or all the objects in the tag queried with `tag.getAllElements()`.
 
-As an example:
+A reminder that `Tag`s contain `Item`s not `ItemStack`s. You can get the `Item` of an `ItemStack` from the stack by calling `<itemStack>.getItem()`.
```
Comment (tterrag1098) [inline docs/utilities/tags.md:31]: This isn't necessary to say.

File: docs/utilities/tags.md [hunk 3]
```diff
@@ -21,17 +21,45 @@ Using Tags In Code
 ------------------
 Block, Item, and Fluid tags are automatically sent from the server to any remote clients on login and reload. Function tags are not synced.
 
+Tags can be accessed in code through the `Tags.Items` and `Tags.Blocks` classes for all the Vanilla Minecraft tags, and through `ResourceLocation`s for ForgeTags.
```
Comment (tterrag1098) [inline docs/utilities/tags.md]: This isn't true, in fact it's backwards. `Tags` (`net.minecraftforge.common.tags.Tags`) holds the *forge* tags. Vanilla tags come from `ItemTags`, `BlockTags`, etc. And you would never access tags "through `ResourceLocation`s".

File: docs/utilities/tags.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 Tags
 ====
-
 Tags are generalized sets of objects in the game, used for grouping related things together and providing fast membership checks.
+Note: Tags have replaced the OreDictionary-- from now on you should (and can only) use tags for all that OreDictionary did. Please see the extra notice at the bottom of this page for more information for transferring from OreDictionary to Tags.
```
Comment (tterrag1098) [inline docs/utilities/tags.md]: Use the admonitions syntax for notes.

File: docs/utilities/tags.md [hunk 5]
```diff
@@ -21,17 +21,45 @@ Using Tags In Code
...
+// There's two methods to do this, doesn't really matter which you choose:
+// Method 1: from the Item's perspective
+inTag = someItemStack.getItem().isIn(tag);
+// Method 2: from the Tag's perspective
+inTag = tag.contains(someItemStack.getItem());
+```
+
+An example of checking if an Item is contained in a certain Forge `Tag`:
+```Java
+// Create a custom resource to use to fetch a tag instance ('mymod:myitemgroup' is the tag)
```
Comment (tterrag1098) [inline docs/utilities/tags.md:45]: This isn't generally correct, the tag should be a static constant somewhere, like all other tags.

File: docs/utilities/tags.md [hunk 6]
```diff
@@ -21,17 +21,45 @@ Using Tags In Code
...
 ResourceLocation myTagId = new ResourceLocation("mymod", "myitemgroup");
+// Tags work through Items, not ItemStacks so we need to get the Item from our ItemStack
 Item unknownItem = stack.getItem();
-boolean isInGroup = ItemTags.getCollection().getOrCreateTag(myTagId).contains(unknownItem);
-// alternatively, can use getTag and perform a null check
+// This line will either retrieve the Tag (referenced by our ResourceLocation) or create it if it is not
+// in the ItemTags collection of all the tags. Then it will check if our Item (again, not ItemStack)
+// exists in the set of Items that belong to this Tag
+boolean isInGroup = ItemTags.getCollection().getOrCreate(myTagId).contains(unknownItem);
+// Alternatively, you can use the getTag and perform a null check to see if the tag exists and do it that way
```
Comment (tterrag1098) [inline docs/utilities/tags.md:53]: This isn't worth mentioning as it should never be done. There is no point, an empty tag and a non-existent tag are treated the same.

File: docs/utilities/tags.md [hunk 7]
```diff
@@ -21,17 +21,45 @@ Using Tags In Code
...
+Tag retrievedTag = ItemTags.getCollection().get(myTagId);
+if(retrievedTag != null){
+    //do something if retrievedTag.contains(unknownItem) or whatever else you'd like to do with that tag
+}
 ```
 
+There is also `ItemTags.getCollection().getOwningTags(item)` that returns all the `Tag`s that contain an `Item`, however this is generally not the way you'd want to handle tags, it's better if you can check if the `Item` is in a specific `Tag` as opposed to retrieving all its `Tag`s and seeing if the `Tag` is in there.
```
Comment (tterrag1098) [inline docs/utilities/tags.md:60]: This is not correct, `IForgeItem#getTags` should be used for this.

**Review by tterrag1098 — CHANGES_REQUESTED** [review]

**Comment by tterrag1098** [comment]
Overall this seems to add a lot of fluff, I'm not sure what the main goal was. The information is quite correct already, this doesn't really correct any misinformation, it just adds some edge cases that are questionably important enough to be in the docs. I'd suggest slimming it down a bit to what's absolutely necessary - more isn't always better.

**Comment by tterrag1098** [comment]
Also, if this information applies to 1.14, it should be targeted there.

File: docs/utilities/tags.md (hunk 6 above)
Comment (Ktar5) [reply docs/utilities/tags.md:53]: I'm not sure what you mean, this is from the previously existing documentation but with extra comments?

Comment (tterrag1098) [reply docs/utilities/tags.md:53]: That doesn't make it correct. If the goal here is to *improve* this page, it should be correct.

**Pull request closed without merging**